src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +208,187 @@ public int draw() {
+                calledVariants++;
+            }
+        }
+        return (double) calledVariants / sampleSize;
+    }
+
+    /**
+     * Simulates the sum of base qualities taken from reads that support the alternate allele by
+     * taking advantage of the fact that the sum of draws from a distribution tends towards a
+     * Gaussian per the Central Limit Theorem.
+     * @param altDepth Number of draws to take from base quality distribution
+     * @param averageQuality Average quality of alt bases
+     * @param standardDeviationQuality Sample standard deviation of base quality scores
+     * @param z number of standard deviation from the mean to take sum over
+     * @return Simulated sum of base qualities the support the alternate allele
+     */
+    static int drawSumOfQScores(final int altDepth, final double averageQuality, final double standardDeviationQuality, final double z) {
+        return (int) (altDepth * averageQuality + z * Math.sqrt(altDepth) * standardDeviationQuality);
+    }
+
+    /**
+     * Calculates the theoretical sensitivity with a given Phred-scaled quality score distribution at a constant
+     * depth.
+     * @param depth Depth to compute sensitivity at
+     * @param qualityHistogram Phred-scaled quality score histogram
+     * @param logOddsThreshold Log odds threshold necessary for variant to be called
+     * @param sampleSize the total number of simulations to run
+     * @param alleleFraction the allele fraction to evaluate sensitivity at
+     * @return Theoretical sensitivity for the given arguments at a constant depth.
+     */
+    private static double sensitivityAtConstantDepth(final int depth, final Histogram<Integer> qualityHistogram, final double logOddsThreshold, final int sampleSize, final double alleleFraction) {
+        return sensitivityAtConstantDepth(depth, qualityHistogram, logOddsThreshold, sampleSize, alleleFraction, RANDOM_SEED);
+    }
+
+    /**
+     * Calculates the theoretical sensitivity with a given Phred-scaled quality score distribution and depth
+     * distribution.
+     * @param depthHistogram Depth histogram to compute theoretical sensitivity over
+     * @param qualityHistogram Phred-scaled quality score histogram
+     * @param sampleSize the total number of simulations to run
+     * @param logOddsThreshold Log odds threshold necessary for variant to be called
+     * @param alleleFraction the allele fraction to evaluate sensitivity at
+     * @return Theoretical sensitivity for the given arguments over a particular depth distribution.
+     */
+    public static double theoreticalSensitivity(final Histogram<Integer> depthHistogram, final Histogram<Integer> qualityHistogram,
+                                                final int sampleSize, final double logOddsThreshold, final double alleleFraction) {
+        if (alleleFraction > 1.0 || alleleFraction < 0.0) {
```

Comment:
If this were GATK code I would suggest `ParamUtils.inRange()`.  Is there a Picard equivalent?

Reply:
There doesn't seem to be a Picard equivalent.

Reply:
Nothing to be done then.